protobuf/src/main/scala/magnolify/protobuf/ProtobufType.scala

```diff
@@ -187,23 +225,45 @@ object ProtobufField {
   implicit val pfByteString = id[ByteString]
   implicit val pfByteArray = aux2[Array[Byte], ByteString](_.toByteArray)(ByteString.copyFrom)
 
-  implicit def pfOption[T](implicit f: ProtobufField[T]): ProtobufField[Option[T]] =
-    new Aux[Option[T], f.FromT, f.ToT] {
-      override val hasOptional: Boolean = true
-      override def from(v: f.FromT): Option[T] = if (v == null) None else Some(f.from(v))
-      override def to(v: Option[T], b: Message.Builder): f.ToT = v match {
-        case None    => null.asInstanceOf[f.ToT]
-        case Some(x) => f.to(x, b)
+  private class OptionProtobufField[T: ClassTag](val f: ProtobufField[T])
+      extends ProtobufField[Option[T]] {
+    override type FromT = f.FromT
+    override type ToT = f.ToT
+    override def from(v: f.FromT): Option[T] = if (v == null) None else Some(f.from(v))
+    override def to(v: Option[T], b: Message.Builder): f.ToT = v match {
+      case None    => null.asInstanceOf[f.ToT]
+      case Some(x) => f.to(x, b)
+    }
+    def typeCheck(v: Any, name: String): Unit = {
```

Comment:
please add doc comment here, separate from my request above to comment the code where this is called